buildbot_nix/repo_config/__init__.py

```diff
@@ -0,0 +1,40 @@
+import tomllib
+from tomllib import TOMLDecodeError
+from typing import TYPE_CHECKING, Self
+
+from buildbot.process.buildstep import ShellMixin
+
+if TYPE_CHECKING:
+    from buildbot.process.log import Log
+from pydantic import BaseModel, ValidationError
+
+
+class RepoConfig(BaseModel):
+    branches: list[str]
+
+
+class BranchConfig(BaseModel):
+    lock_file: str = "flake.lock"
+    attribute: str = "checks"
+
+    @classmethod
+    async def extract_during_step(cls, buildstep: ShellMixin) -> Self:
+        stdio: Log = await buildstep.addLog("stdio")
+        cmd = await buildstep.makeRemoteShellCommand(
+            collectStdout=True,
+            collectStderr=False,
+            stdioLogName=None,
+            command=["cat", "buildbot-nix.toml"],
```

Comment:
```suggestion
            # TODO: replace this with something like buildbot.steps.transfer.StringUpload in the future... this one doesn't not exist yet.
            command=["cat", "buildbot-nix.toml"],
```

Reply:
Can you make a table in our Readme with the current configured values?

Reply:
Will do!